From 2099dd4a35a444ba6185df302cfb7428074acf8b Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 22 Nov 2023 15:51:24 +0100
Subject: [PATCH 453/841] feat(stm32mp2): skip ICN configuration if M33 is
 TDCID

If Cortex-M33 is TDCID, Cortex-A35 does not have access to Interconnect
configuration registers. Put stm32mp2_syscfg_set_icn_qos() call under
!STM32MP_M33_TDCID flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 314555c191cf55be666207617df335180c7ffa35 feat(stm32mp2): skip ICN configuration if M33 is TDCID]
Change-Id: Id64089f0bc3e7e2ef6583ed6816ee5f72d386e50
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/347833
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index aa7e53f13..5a42e631d 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -213,8 +213,10 @@ void bl2_platform_setup(void)
 		panic();
 	}
 
+#if !STM32MP_M33_TDCID
 	/* Set QOS ICN priority */
 	stm32mp2_syscfg_set_icn_qos();
+#endif
 }
 
 static void reset_backup_domain(void)
-- 
2.47.2

